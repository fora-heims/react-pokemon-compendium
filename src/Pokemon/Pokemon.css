.pokemon {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-family: 'Courier New', Courier, monospace;
  font-size: 12px;
  border: 3px yellowgreen solid;
  color: black;
  border-radius: 5px;
  width: 130px;
  height: 200px;
  text-decoration: none;
}

.name {
  font-size: 16px;
  font-weight: 900;
}

img {
  width: 110px;
}

.pokemon:hover {
  filter: drop-shadow(8px 8px 8px orange);
}

img:hover {
  scale: 20%;
}
