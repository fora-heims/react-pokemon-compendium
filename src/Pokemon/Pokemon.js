import React from 'react';
import './Pokemon.css';

export default function Pokemon(props) {
  return (
    <a
      className="pokemon"
      rel="noreferrer"
      target="_blank"
      href={`https://www.google.com/search?q=${props.pokemon}`}
    >
      <img src={props.url_image} alt={props.pokemon} />
      <span className="name">{props.pokemon}</span>
      <span>Type: {props.type_1}</span>
      <span>HP: {props.hp}</span>
      <span>Attack: {props.attack}</span>
      <span>Defense: {props.defense}</span>
    </a>
  );
}
