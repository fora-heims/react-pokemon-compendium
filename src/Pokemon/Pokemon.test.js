import { render } from '@testing-library/react';
import Pokemon from './Pokemon.js';

test('', () => {
  const container = render(
    <Pokemon
      pokemon="considsrtuhsrthsxaw"
      hp="110"
      attack="20"
      defense="40"
      type_1="person"
      url_image="http://assets.pokemon.com/assets/cms2/img/pokedex/full/172.png"
    />
  );
  expect(container).toMatchSnapshot();
});
