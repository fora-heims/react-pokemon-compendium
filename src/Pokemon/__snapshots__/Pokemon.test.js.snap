// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="pokemon"
        href="https://www.google.com/search?q=considsrtuhsrthsxaw"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="considsrtuhsrthsxaw"
          src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/172.png"
        />
        <span
          class="name"
        >
          considsrtuhsrthsxaw
        </span>
        <span>
          Type: 
          person
        </span>
        <span>
          HP: 
          110
        </span>
        <span>
          Attack: 
          20
        </span>
        <span>
          Defense: 
          40
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="pokemon"
      href="https://www.google.com/search?q=considsrtuhsrthsxaw"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="considsrtuhsrthsxaw"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/172.png"
      />
      <span
        class="name"
      >
        considsrtuhsrthsxaw
      </span>
      <span>
        Type: 
        person
      </span>
      <span>
        HP: 
        110
      </span>
      <span>
        Attack: 
        20
      </span>
      <span>
        Defense: 
        40
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
